### PR TITLE
fix: css and html issues

### LINKED ISSUE #41921: Broken Test Case Regex
This issue is an extension of #41905. Multiple challenges suffer the same regex incorrectness in CSS challenges. This is a list of affected challenges: 
- [ ] [Specify How Fonts Should Degrade](https://www.freecodecamp.org/learn/responsive-web-design/basic-css/specify-how-fonts-should-degrade)
- [ ] [Size Your Images](https://www.freecodecamp.org/learn/responsive-web-design/basic-css/size-your-images)
- [ ] [Use Abbreviated Hex Code](https://www.freecodecamp.org/learn/responsive-web-design/basic-css/use-abbreviated-hex-code)
- [ ] [Use RGB to Mix Colors](https://www.freecodecamp.org/learn/responsive-web-design/basic-css/use-rgb-to-mix-colors)
- [ ] [ES6 Template Literals](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals)

**Expected behavior**
Test cases should pass, but the regex incorrectness in handling spaces prevents it from passing. 

**System (please complete the following information):**
 - Device: All
 - OS: All
 - Browser: Chrome, Firefox
 - Version: Latest

**Additional context**
Refer to #41905 for more info. I can work on this.

### PR DESCRIPTION
**Pull request by SethFalco**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #41921

Trims the class attr on HTML elements before assertion.

> A string containing a set of space-separated tokens may have leading or trailing ASCII whitespace.
> \- https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#set-of-space-separated-tokens

Allows whitespaces before the colon between the CSS property and value.

> Both properties and values are case-insensitive by default in CSS. The pair is separated by a colon, ':' (U+003A COLON), and white spaces before, between, and after properties and values, but not necessarily inside, are ignored.
> \- https://developer.mozilla.org/en-US/docs/Web/CSS/Syntax#css_declarations

Makes the semi-colon of tests that test a block with a single property optional.

> The last declaration of a block doesn't need to be terminated by a semi-colon, though it is often considered good style to do it as it prevents forgetting to add it when extending the block with another declaration.
> \- https://developer.mozilla.org/en-US/docs/Web/CSS/Syntax#css_declarations

Previously tests that checked if an HTML element had a class would only check the literal value of the attribute. That behavior makes the test fail if the user added multiple classes to the element. This checks if it has _at least_ the target class, and still passes if the element has other classes.

### CODE REVIEW

**Comment by SethFalco** [comment]
Just a few more changes specific to "Specify How Fonts Should Degrade", which I missed.

Previously `"Lobster'`, or `Lobster'` were acceptable for this test, now it must be one of:
* `Lobster`
* `"Lobster"`
* `'Lobster'`

Allow whitespace before the comma.

File: curriculum/challenges/english/01-responsive-web-design/basic-css/give-a-background-color-to-a-div-element.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ assert($('div').css('background-color') === 'rgb(192, 192, 192)');
 A class named `silver-background` should be defined within the `style` element and the value of `silver` should be assigned to the `background-color` property.
 
 ```js
-assert(code.match(/\.silver-background\s*{\s*background-color:\s*silver;\s*}/));
+assert(code.match(/\.silver-background\s*{\s*background-color\s*:\s*silver;?\s*}/));
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/01-responsive-web-design/basic-css/give-a-background-color-to-a-div-element.md:43]:
```suggestion
assert(__helpers.removeWhiteSpace(code).match(/\.silver-background{background-color:silver;?}/));
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Hey there,

Thank you, for opening a PR. Would you mind doing similarly to the other tests? It is just nicer to look at, and also allows for cases where there is a space between semi-colons `;`

**Comment by SethFalco** [comment]
Think it's worth updating all other impacted tests to use `__helper.remoteWhiteSpace` as well?
I assume I could just find the issue via regex and change 'em all?

image

**Comment by ShaunSHamilton** [comment]
> Think it's worth updating all other impacted tests to use __helper.removeWhiteSpace as well?

In general, _if it is not broken, do not fix it_. The reason I suggested the change in this PR is the files were/are being worked on.

However, yesterday Tom pointed out how, most of the time, `removeWhiteSpace` is worse than just being as specific as possible with Regex, because:
```
assert(code.match(/\.some-class\s*{/)); // .some-class { - passes
assert(__helpers.removeWhiteSpace(code).match(/\.some-class{/)); // .so me - clas s { - also passes
```
Ideally, we should be using some sort of `trim`/`split` function for another `removeWhiteSpace` function.

---

So, I will wait for another to comment on whether we should completely ditch this use-case for `removeWhiteSpace`. Sorry about the hassle.
```

**Comment by SethFalco** [comment]
> .so me - clas s { - also passes

Ahh right, that didn't occur to me. ^-^'

Just to put my opinion out, with the above in mind I think it's best not to use `removeWhiteSpace` here.

**Comment by Sembauke** [comment]
So seeing Shaun's comment pointing out what Tom said I would suggest changing it back to using RegEx and only change these challenges for now and wait for #41942

**Review by moT01 — DISMISSED** [review]
LGTM @SethFalco 🎉 

If the only thing stopping this from closing the issue is the ES6 Template Literal challenge, I left a comment on the issue - I think it would be fine to leave that one, and this could close the issue. Other opinions are welcome.

**Review by moT01 — COMMENTED** [review]
Actually, I noticed that the issue has [Use RGB to mix colors](https://www.freecodecamp.org/learn/responsive-web-design/basic-css/use-rgb-to-mix-colors) on it, but it's not included in this PR. I'm not sure if I missed a comment about it or perhaps the hex one was mistaken for it. Want to take a quick look at that @SethFalco

**Comment by SethFalco** [comment]
> Actually, I noticed that the issue has [Use RGB to mix colors](https://www.freecodecamp.org/learn/responsive-web-design/basic-css/use-rgb-to-mix-colors) on it, but it's not included in this PR. I'm not sure if I missed a comment about it or perhaps the hex one was mistaken for it. Want to take a quick look at that @SethFalco

Good catch! Sorry about that, we should be cool now, but we'll see if CI agrees.

File: curriculum/challenges/english/01-responsive-web-design/basic-css/understand-absolute-versus-relative-units.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ assert(
 Your `red-box` class should give elements 1.5em of `padding`.
 
 ```js
-assert(code.match(/\.red-box\s*?{[\s\S]*padding:\s*?1\.5em/gi));
+assert(code.match(/\.red-box\s*?{[\s\S]*padding\s*:\s*?1\.5em/gi));
```
Comment (Sembauke) [inline curriculum/challenges/english/01-responsive-web-design/basic-css/understand-absolute-versus-relative-units.md:40]: Hey @SethFalco, this one also seems to miss a semi-colon.

**Review by Sembauke — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/01-responsive-web-design/basic-css/understand-absolute-versus-relative-units.md (hunk 2 above)
Comment (SethFalco) [reply curriculum/challenges/english/01-responsive-web-design/basic-css/understand-absolute-versus-relative-units.md:40]: Matching an optional semi-colon or not matching a semi-colon at all would result in the same behavior.
The last item in a CSS rule doesn't have to have a semi-colon, which is where the user is encouraged to add it, so this is valid.

```css
.red-box {
  background-color: red;
  margin: 20px 40px 20px 40px;
  padding: 1.5em
}
```

I can change it if you want, but _in my opinion_ for as long as we're using regex, I think this is better. Once we move to a CSS parser/helper, then maybe we can be more pedantic about it?

**Review by Sembauke — APPROVED** [review]
LGTM 🥳

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍

**Pull request merged**